src/netdev.c

```diff
@@ -94,52 +94,71 @@ ni_netdev_slaveinfo_destroy(ni_slaveinfo_t *slave)
 	memset(slave, 0, sizeof(*slave));
 }
 
-static void
-ni_netdev_free(ni_netdev_t *dev)
+void
+ni_netdev_reset(ni_netdev_t *dev)
 {
-	/* Clear out linkinfo */
-	ni_string_free(&dev->link.qdisc);
-	ni_string_free(&dev->link.kind);
+	if (!dev)
+		return;
+
+	/* Clear and Reset all (kernel) property data,
+	 * except of list, ref counter & marker flags */
```

Comment:
Just curious, what about:
```
dev->seq
dev->modifed
dev->deleted
dev->created

Reply:
This is same as with saved_mtu -- it's a state flag we use when we process events in **wickedd**
and not a property (e.g. when we send NEWLINK creating a new  device, we set dev->created
until a NEWLINK event from kernel arrives that "updates" it -> dev->created causes to send a
NI_EVENT_DEVICE_CREATE dbus event then in contrast to NI_EVENT_DEVICE_CHAGED
or other events --> all arriving as NEWLINK from kernel).

The reset is done in the fsm (nanny/client side): we need to reset (ni_netdev_t)dbus-object->handle
before we apply properties to it. When wickedd sends us a message with e.g. master "br0" reference
property, we set the dev->link.masterdev.name/index. Then it gets unenslaved in kernel, there is a
NEWLINK which resets  dev->link.masterdev.name/index. In the dbus layer, the dbus-propery->get()
returns no_property for "master" property in the client side msg does not contain any -- this caused,
that the dev->link.masterdev.name/index were not reset from "br0" to NULL.

Reply:
ic, thx!